Volume/Analytics/Analytics.swift

```diff
@@ -9,11 +9,82 @@
 import AppDevAnalytics
 
 // volume-specific extensions of base Event protocol
-
 struct StartOnboarding: Event {
     let name = "start_onboarding"
 }
 
 struct CompleteOnboarding: Event {
     let name = "complete_onboarding"
 }
+
+struct FollowPublication: Event {
+    let name = "follow_publication"
+    var parameters: [String : Any]?
+}
+
+struct UnfollowPublication: Event {
+    let name = "unfollow_publication"
+    var parameters: [String : Any]?
+}
+
+struct OpenArticle: Event {
+    let name = "open_article"
+    var parameters: [String : Any]?
+}
+
+struct CloseArticle: Event {
+    let name = "close_article"
+    var parameters: [String : Any]?
+}
+
+struct ShareArticle: Event {
+    let name = "share_article"
+    var parameters: [String : Any]?
+}
+
+struct ShoutoutArticle: Event {
+    let name = "shoutout_article"
+    var parameters: [String : Any]?
+}
+
+struct BookmarkArticle: Event {
+    let name = "bookmark_article"
+    var parameters: [String : Any]?
+}
+
+struct UnbookmarkArticle: Event {
+    let name = "unbookmark_article"
+    var parameters: [String : Any]?
+}
+
+// Helper struct and enums to disambiguate parameters for different event types
+struct Parameters {
+    static func params(for event: EventLevel, id: String, at entry: EntryPoint) -> [String: Any] {
```

Comment:
Could be more concise/idiomatic as 

```suggestion
    static func for(_ event: EventLevel, id: String, at entry: EntryPoint) -> [String: Any] {
```

And then call with

```
Parameters.for(.article, id: id, entry: .articleDetail)
```

Reply:
for is reserved